### PR TITLE
Create initial PUT /user test

### PR DESCRIPTION
**Pull request by jeremyabbott**
@crookedneighbor how is this for an initial test (test/api/user/PUT-user.test.js)? Thank you again for the help!

### CODE REVIEW

File: test/api/user/PUT-user.test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import {
+  generateUser,
+  requester,
+} from '../../helpers/api.helper';
+
+describe('PUT /user', () => {
+  let api, user;
+  
+  beforeEach((done) => {
+    generateUser().then((usr) => {
```
Comment (crookedneighbor) [inline test/api/user/PUT-user.test.js]: the beforeEach block can return a promise as well:

``` js
let user;

beforeEach(() => {
  return generateUser().then((usr) => {
    user = usr;
  });
});
```

File: test/api/user/PUT-user.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      user = usr;
+      api = requester(usr);
+      done();
+    }).catch(done);
+  });
+  
+  it("updates the user", () => {
```
Comment (crookedneighbor) [inline test/api/user/PUT-user.test.js]: We try to use single quotes in our javascript whenever possible.

**Comment by jeremyabbott** [comment]
@crookedneighbor updated with feedback. Thank you for your patience too! I really appreciate it.

**Comment by crookedneighbor** [comment]
I made some adjustments and added some more tests. Thanks for getting this started. https://github.com/HabitRPG/habitrpg/blob/develop/test/api/user/PUT-user.test.js

This will count towards a contributor tier! What's your User ID? You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Comment by jeremyabbott** [comment]
Wow! Thanks for the feedback. It's helping me grok the idea of protected paths concept. Is it cool if I keep working on tests?

My user ID is: f2fdbe8a-25d9-4c38-933f-7a6ff5885062

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

Feel free to keep adding tests if you'd like. The dream is to move over all the tests from the legacy api tests so we can remove that directory entirely and provide coverage for every route. 

The only ones that you shouldn't work on are the ones related to tasks, as I'll be doing that as part of a big tasks collection PR.

**Comment by jeremyabbott** [comment]
Thank you!

I started looking at the legacy tests this morning. I'll start replacing
those. Thank you again.

On Mon, Oct 26, 2015 at 9:40 PM, Blade Barringer notifications@github.com
wrote:

> Tier awarded! You can read more about contributor tiers and rewards here
> http://habitrpg.wikia.com/wiki/Contributor_Rewards.
> 
> Note that as your tier increases, it takes progressively more work to
> achieve the next one, so for the higher tiers, larger PRs or a few PRs are
> required. There's some information about that at Contributor Rewards, but
> feel free to ask here if you have questions!
> 
> Feel free to keep adding tests if you'd like. The dream is to move over
> all the tests from the legacy api tests so we can remove that directory
> entirely and provide coverage for every route.
> 
> The only ones that you shouldn't work on are the ones related to tasks, as
> I'll be doing that as part of a big tasks collection PR.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/HabitRPG/habitrpg/pull/6107#issuecomment-151351141.

## 

Thanks,
Jeremy

---

Q: Why is this email three sentences or less?
A: http://three.sentenc.es

**Pull request merged**